# El enigma de la Flor Matakiterani: Desentrañando patrones matemáticos

En una misteriosa y enigmática isla, florece una planta única en el mundo: la Flor Matakiterani. Esta flor, venerada por los isleños, es famosa por su belleza excepcional y su patrón único de crecimiento en el número de pétalos. Según las leyendas locales, la Flor Matakiterani posee propiedades mágicas y curativas. Los habitantes de la isla han observado que estas flores pueden producir desde un mínimo de $8$ pétalos y hasta un asombroso máximo de $1502$ pétalos por día. En este taller, exploraremos las propiedades matemáticas de esta flor. 

1. La Flor Matakiterani sigue una secuencia específica en el número de pétalos que la flor produce cada día. Después de años de observación, los isleños han descubierto que la cantidad de pétalos que la flor produce en el $n$-ésimo día sigue la fórmula: $F_{n} = 6n + 2$, dónde $n$ corresponde al número de días. Calcula cuántos pétalos producirá la Flor Matakiterani en el día $12$.
1. ¿Cuántos días deben transcurrir para que la flor alcance su valor máximo de pétalos producidos?
1. ¿Existe un día en que la flor Matakiterani produzca $912$ pétalos? Explique.
1. Los isleños descubrieron otra flor cuyas propiedades parecen estar ligadas a otro tipo de secuencia. Durante los primeros $8$ días, esta flor crece de manera que el número de pétalos producidos en el $n$-ésimo día se duplica con respecto al día anterior. Si el primer día produce $5$ pétalos, construye la expresión algebraica $G_{n}$ que represente el número de pétalos producidos en el $n$-ésimo día e imprime ordenadamente la cantidad de pétalos producidos durante los primeros ocho días.
1. Los isleños han notado que, luego de los primeros $8$ días, la magia de esta segunda flor pierde potencia, haciendo que el crecimiento sea más lento, modelado por la expresión: $$H_n = 4 \cdot (n - 9)\text{ para }n > 8.$$ Calcula el total de pétalos producidos por esta segunda flor al finalizar un mes de $31$ días, considerando que comienza a florecer el día $1$.


In [1]:
from scipy.optimize import fsolve

---

### Respuestas

#### 1.

In [2]:
def petalos_flor(dia: int | float) -> float:
    return min(6 * dia + 2, 1502)

In [3]:
res = petalos_flor(12)
print(f"Al duodécimo día la flor tendrá {res} pétalos.")

Al duodécimo día la flor tendrá 74 pétalos.


#### 2.

In [4]:
res = fsolve(lambda dia: petalos_flor(dia) - 1502, 200)[0]
print(f"La flor alcanza su máximo al día: {res:n}.")

La flor alcanza su máximo al día: 250.


#### 3.

In [5]:
res = fsolve(lambda dia: petalos_flor(dia) - 912, 100)[0]
print(f"La flor alcanza 912 pétalos el día: {res:n}.")
print(f"Pétalos el día 151: {petalos_flor(151)}")
print(f"Pétalos el día 152: {petalos_flor(152)}")

La flor alcanza 912 pétalos el día: 151.667.
Pétalos el día 151: 908
Pétalos el día 152: 914


Respuesta:

No, el día $151$ la flor tendrá $908$ y al siguiente $914$

#### 4.

$a_n = a_1 · r^{(n-1)}$

$a_1 = 5$

$r = 2$

> $G(n)=5·2^{(n-1)}$

In [6]:
def flor_primeros_8_dias(dia: int) -> int:
    return 5 * 2 ** (dia - 1)

for dia in range(1, 8 + 1):
    print(f"Día {dia}:\t{flor_primeros_8_dias(dia):>3n}")

Día 1:	  5
Día 2:	 10
Día 3:	 20
Día 4:	 40
Día 5:	 80
Día 6:	160
Día 7:	320
Día 8:	640


#### 5.

In [7]:
def flor_desde_dia_9(dia: int) -> int:
    return 4 * (dia - 9)

In [8]:
petalos_primeros_8_dias = sum([flor_primeros_8_dias(dia) for dia in range(1, 9)])
petalos_dias_siguientes = sum([flor_desde_dia_9(dia) for dia in range(9, 32)])
res = petalos_primeros_8_dias + petalos_dias_siguientes

print(f"Total pétalos primeros 8 días: {petalos_primeros_8_dias}.")
print(f"Total pétalos días 9 al 31: {petalos_dias_siguientes}.")
print(f"\nEl total de pétalos al finalizar el día 31 es de: {res}")

Total pétalos primeros 8 días: 1275.
Total pétalos días 9 al 31: 1012.

El total de pétalos al finalizar el día 31 es de: 2287


_Revisamos coherencia de los resultados:_

In [10]:
for dia in range(1, 32):
    if dia < 9:
        print(f"día {dia}:\t{flor_primeros_8_dias(dia)}")
    else:
        print(f"día {dia}:\t{flor_desde_dia_9(dia)}")

día 1:	5
día 2:	10
día 3:	20
día 4:	40
día 5:	80
día 6:	160
día 7:	320
día 8:	640
día 9:	0
día 10:	4
día 11:	8
día 12:	12
día 13:	16
día 14:	20
día 15:	24
día 16:	28
día 17:	32
día 18:	36
día 19:	40
día 20:	44
día 21:	48
día 22:	52
día 23:	56
día 24:	60
día 25:	64
día 26:	68
día 27:	72
día 28:	76
día 29:	80
día 30:	84
día 31:	88
